### PR TITLE
Python: Flask: Identify body contents passed via named response parameter in invocations of Response constructor

### PR DESCRIPTION
**Pull request by zbazztian**
As witnessed [here](https://github.com/Netflix/security_monkey/blob/develop/security_monkey/auth/modules.py#L266).

Note that Flask's [Response](https://github.com/pallets/flask/blob/main/src/flask/wrappers.py#L136) class inherits from `werkzeug`'s [Response](https://github.com/pallets/werkzeug/blob/main/src/werkzeug/wrappers/response.py#L172) class.

### CODE REVIEW

File: python/ql/test/library-tests/frameworks/flask/response_test.py [hunk 1]
```diff
@@ -57,6 +57,11 @@ def html7():  # $requestHandler
     resp.set_data("<h1>hello</h1>")  # $ MISSING: responseBody="<h1>hello</h1>"
     return resp  # $ SPURIOUS: HttpResponse mimetype=text/html responseBody=resp
 
+@app.route("/html8")  # $routeSetup="/html8"
+def html8():  # $requestHandler
+    resp = Response(response="<h1>hello</h1>")  # $HttpResponse mimetype=text/html responseBody="<h1>hello</h1>"
+    return resp  # $ SPURIOUS: HttpResponse mimetype=text/html responseBody=resp
+
```
Comment (yoff) [inline python/ql/test/library-tests/frameworks/flask/response_test.py:64]: I would expect this test case to appear right after `html4`, which, ironically, would entail a bunch of renames...

**Review by yoff — CHANGES_REQUESTED** [review]
I am going to be annoying and ask you to rearrange the code slightly, but otherwise LTGM.
And thanks for fixing this oversight!

File: python/ql/test/library-tests/frameworks/flask/response_test.py (hunk 1 above)
Comment (yoff) [reply python/ql/test/library-tests/frameworks/flask/response_test.py:64]: (I seem unable to construct a suggestion for this..)

Comment (zbazztian) [reply python/ql/test/library-tests/frameworks/flask/response_test.py:64]: Haha, ok, will do.

**Comment by zbazztian** [comment]
Ok, addressed. I also squashed everything into a single commit.

**Review by yoff — DISMISSED** [review]
Thanks for the fix.

File: python/ql/lib/change-notes/2022-03-30-flask-recognize-body-param.md [hunk 2]
```diff
@@ -0,0 +1,4 @@
+---
+category: minorAnalysis
+---
+* The body content of those Flask `Response` objects which were constructed by explicitly referring to the `response` parameter is now detected.
```
Comment (RasmusWL) [inline python/ql/lib/change-notes/2022-03-30-flask-recognize-body-param.md:4]:
```suggestion
* Improved modeling of Flask `Response` objects, so passing a response body with the keyword argument `response` is now recognized.
```

**Review by RasmusWL — DISMISSED** [review]
Thanks for doing this :+1: I have a minor suggestion for the change-note

**Comment by zbazztian** [comment]
@RasmusWL Thanks, I committed your suggestion.

**Review by RasmusWL — APPROVED** [review]

**Pull request merged**